### PR TITLE
Add API throttling support

### PR DESCRIPTION
**Pull request by edudepetris**
### What should this PR do?
Solve issue #56 

### Solution
Introduces a new `ExceedRateLimit` exception, and raise it when the response status code is 429.

Notice that running this script it will return the new execption.

```rb
require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  gem 'concurrent-ruby', require: 'concurrent'
  gem 'harvesting', github: 'edudepetris/harvesting', branch: 'ed/429-throttled'
end

client = Harvesting::Client.new(
  access_token: <your-token-here>,
  account_id: <your-account-id>
)

promises = (1..200).map do
  Concurrent::Promise.execute do
    client.me
  end
end

Concurrent::Promise.zip(*promises).value!
```

### CODE REVIEW

File: lib/harvesting/errors.rb [hunk 1]
```diff
@@ -4,4 +4,7 @@ class AuthenticationError < StandardError
 
   class UnprocessableRequest < StandardError
   end
+
+  class ExceedRateLimit < StandardError
```
Comment (etagwerker) [inline lib/harvesting/errors.rb:8]: Could you rename this class as `RateLimitExceeded`?

**Review by etagwerker — CHANGES_REQUESTED** [review]
@edudepetris Looks great! Just added one comment. 💯

File: lib/harvesting/errors.rb (hunk 1 above)
Comment (edudepetris) [reply lib/harvesting/errors.rb]: updated 2c2a2d2bd3a69620e1dee3e49809626e6052b15a

**Review by etagwerker — APPROVED** [review]
Great, thanks! 👍

**Pull request merged**